src/unity/python/turicreate/visualization/show.py

```diff
@@ -96,46 +96,302 @@ def show(x, y, xlabel="X", ylabel="Y", title=None):
+
+    Examples
+    --------
+    Make a heatmap.
+
+    >>> x = turicreate.SArray([1,2,3,4,5])
+    >>> y = x * 2 
+    >>> heat = turicreate.visualization.heatmap(x, y)
+    """
+    if (type(x) != tc.data_structures.sarray.SArray or 
+        type(y) != tc.data_structures.sarray.SArray or
+        x.dtype not in [int, float] or y.dtype not in [int, float]):
+        raise ValueError("turicreate.visualization.heatmap supports " + 
+            "SArrays of dtype: int, float")
     path_to_client = _get_client_app_path()
     title = _get_title(title)
     plt_ref = tc.extensions.plot_heatmap(path_to_client, x, y, 
       xlabel, ylabel, title)
     return Plot(plt_ref)
 
 def box_plot(x, y, xlabel="X", ylabel="Y", title=None):
+    """
+    Plots the data in `x` on the X axis and the data in `y` on the Y axis
+    in a 2d box and whiskers plot, and returns the resulting Plot object.
+    
+    The function x as SArray of dtype str and y as SArray of dtype: int, float.
+
+    Parameters
+    ----------
+    x : SArray
+      The data to plot on the X axis of the box and whiskers plot. 
+      Must be an SArray with dtype string
+    y : SArray
+      The data to plot on the Y axis of the box and whiskers plot. 
+      Must be numeric (int/float) and must be the same length as `x`.
+    xlabel : str (optional)
+      The text label for the X axis. Defaults to "X".
+    ylabel : str (optional)
+      The text label for the Y axis. Defaults to "Y".
+    title : str (optional)
+      The title of the plot. Defaults to None. If the value is None, the title
+      will be "<xlabel> vs. <ylabel>". Otherwise, the string passed in as the
+      title will be used as the plot title.
+
+    Return Value
+    ------------
+    An object of Plot that is the box and whiskers plot
+
+    Examples
+    --------
+    Make a box and whiskers plot.
+
+    >>> bp = turicreate.visualization.box_plot(tc.SArray(['a','b','c','a','a']),tc.SArray([4.0,3.25,2.1,2.0,1.0]))
+    """
+    if (type(x) != tc.data_structures.sarray.SArray or 
+        type(y) != tc.data_structures.sarray.SArray or
+        x.dtype != str or y.dtype not in [int, float]):
+        raise ValueError("turicreate.visualization.box_plot supports " + 
+            "x as SArray of dtype str and y as SArray of dtype: int, float." +
+            "\nExample: turicreate.visualization.box_plot(tc.SArray(['a','b','c','a','a']),tc.SArray([4.0,3.25,2.1,2.0,1.0]))")
     path_to_client = _get_client_app_path()
     title = _get_title(title)
     plt_ref = tc.extensions.plot_boxes_and_whiskers(path_to_client, x, y, 
       xlabel, ylabel, title)
     return Plot(plt_ref)
 
 def columnwise_summary(sf):
+    """
+    Plots a columnwise summary of the sframe provided as input, 
+    and returns the resulting Plot object.
+    
+    The function supports SFrames.
+
+    Parameters
+    ----------
+    sf : SFrame
+      The data to get a columnwise summary for
+    
+    Return Value
```

Comment:
Same comment as for scatter plot (should be Returns, with name/type/description in special syntax).